### PR TITLE
Add include guards to all *.h files

### PR DESCRIPTION
**Pull request by masterleinad**

### CODE REVIEW

File: include/deal.II/base/std_cxx1x/mutex.h [hunk 1]
```diff
@@ -13,10 +13,15 @@
...
+#define dealii__std_cxx1x_mutex_h
+
 // this file is deprecated. simply include the one we use now
 #include "../std_cxx11/mutex.h"
 
 // then allow using the old namespace name instead of the new one
 DEAL_II_NAMESPACE_OPEN
 namespace std_cxx1x = std_cxx11;
-DEAL_II_NAMESPACE_CLOSE
+iDEAL_II_NAMESPACE_CLOSE
```
Comment (davydden) [inline include/deal.II/base/std_cxx1x/mutex.h]: typo

File: include/deal.II/base/tensor_base.h [hunk 2]
```diff
@@ -1,6 +1,7 @@
 #ifndef dealii__tensor_base_h
 #define dealii__tensor_base_h
 #warning This file is deprecated. Use <deal.II/base/tensor.h> instead.
-#endif
 
 #  include <deal.II/base/tensor.h>
+
+#endif
```
Comment (davydden) [inline include/deal.II/base/tensor_base.h]: as long as there is guard in `base/tensor.h`, this should not make a difference.

Comment (masterleinad) [inline include/deal.II/base/tensor_base.h]: Sure, the same would be true for all `include/deal.II/base/std_cxx1x/*.h` files. 
I just consider it good style to have these guards in all header files.

**Comment by drwells** [comment]
I am surprised how many of these were missing. Thanks!

**Comment by bangerth** [comment]
Yes, surprising indeed!

The way to automatically test this would have been to `#include` the same file twice in those tests that we run for each header file that makes sure a header is compilable by itself. Anyone want to make that change in the corresponding cmake scripts?

**Pull request merged**